From afd46d28b506d1446f54c90d914eb4fa7dff6ae6 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Mon, 8 Jul 2013 11:21:44 +0200
Subject: [PATCH 264/628] spi: xilinx-qps: Fix PM ops

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 8b32c67034ece2fde9e1c37c644c8c0d950cd1f9

Use SIMPLE_DEV_PM_OPS macro instead of ifdef mess.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/spi/spi-xilinx-qps.c |   11 +++--------
 1 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/drivers/spi/spi-xilinx-qps.c b/drivers/spi/spi-xilinx-qps.c
index c48dbd5..4d19871 100644
--- a/drivers/spi/spi-xilinx-qps.c
+++ b/drivers/spi/spi-xilinx-qps.c
@@ -997,15 +997,10 @@ static int xqspips_resume(struct device *dev)
 	dev_dbg(&pdev->dev, "resume succeeded\n");
 	return 0;
 }
-static const struct dev_pm_ops xqspips_dev_pm_ops = {
-	SET_SYSTEM_SLEEP_PM_OPS(xqspips_suspend, xqspips_resume)
-};
-#define XQSPIPS_PM	(&xqspips_dev_pm_ops)
-
-#else /* ! CONFIG_PM_SLEEP */
-#define XQSPIPS_PM	NULL
 #endif /* ! CONFIG_PM_SLEEP */
 
+static SIMPLE_DEV_PM_OPS(xqspips_dev_pm_ops, xqspips_suspend, xqspips_resume);
+
 /**
  * xqspips_probe - Probe method for the QSPI driver
  * @pdev:	Pointer to the platform_device structure
@@ -1210,7 +1205,7 @@ static struct platform_driver xqspips_driver = {
 		.name = DRIVER_NAME,
 		.owner = THIS_MODULE,
 		.of_match_table = xqspips_of_match,
-		.pm = XQSPIPS_PM,
+		.pm = &xqspips_dev_pm_ops,
 	},
 };
 
-- 
1.7.5.4

